src/main/java/org/reaktivity/nukleus/socks/internal/stream/SocksServerFactory.java

```diff
@@ -539,27 +518,26 @@ private void doSocksHandshakeReply(
             SocksAuthenticationMethod method)
         {
             SocksHandshakeReplyFW handshakeReply = handshakeReplyRW.wrap(writeBuffer,
-                DataFW.FIELD_OFFSET_PAYLOAD,
-                writeBuffer.capacity())
-                .version(5)
-                .method(method.value())
-                .build();
+                                                                         DataFW.FIELD_OFFSET_PAYLOAD,
+                                                                         writeBuffer.capacity())
+                                                                   .version(5)
+                                                                   .method(method.value())
+                                                                   .build();
 
             doNetworkData(handshakeReply.buffer(), handshakeReply.offset(), handshakeReply.sizeof());
         }
 
         private void doSocksCommandReply(
-            String address,
+            SocksAddressFW address,
             int port)
         {
-            byte[] ipv4Address = lookupName(address).getAddress();
             SocksCommandReplyFW socksCommandReply = socksCommandReplyRW.wrap(writeBuffer,
                 DataFW.FIELD_OFFSET_PAYLOAD,
                 writeBuffer.capacity())
                 .version(5)
                 .type(t -> t.set(SocksCommandReplyType.SUCCEEDED))
                 .reserved(0)
-                .address(a -> a.ipv4Address(i -> i.set(ipv4Address)))
+                .address(s -> s.ipv4Address(i -> i.set(address.ipv4Address())))
```

Comment:
I think we changed `SocksAddress` to be binary compatible, so no need to assume `IPv4` here, just set the whole address directly, right?

Reply:
what do you mean by set the whole address? if I do `s -> s.ipv4Address(i -> i.set(address))`, then this line will become an error `cannot resolve method set( ... socksAddressFw)`

Reply:
After I changed to `SocksNetworkAddressFW`, still cannot setup as `i.set(address)`, any suggestions?

Reply:
done with helper method.